src/mcp_atlassian/servers/jira.py

```diff
@@ -159,10 +159,361 @@ async def get_issue(
+                                .get("name"),
+                            }
+                            result["pull_requests"].append(pr_info)
+
+                    # Process commits
+                    if "commits" in detail:
+                        for commit in detail["commits"]:
+                            commit_info = {
+                                "id": commit.get("id"),
+                                "displayId": commit.get("displayId"),
+                                "message": commit.get("message"),
+                                "author": commit.get("author", {}).get("name"),
+                                "authorTimestamp": commit.get("authorTimestamp"),
+                                "url": commit.get("url"),
+                                "repository": commit.get("repository", {}).get("name"),
+                            }
+                            result["commits"].append(commit_info)
+
+            # Add summary information
+            result["summary"] = {
+                "total_branches": len(result["branches"]),
+                "total_repositories": len(result["repositories"]),
+                "total_pull_requests": len(result["pull_requests"]),
+                "total_commits": len(result["commits"]),
+            }
+
+            # Only return the result if we found any development data
+            if any(
+                [
+                    result["branches"],
+                    result["repositories"],
+                    result["pull_requests"],
+                    result["commits"],
+                ]
+            ):
+                logger.debug(
+                    f"✅ Found dev status for {issue_key}: {result['summary']}"
+                )
+                return result
+            else:
+                logger.debug(f"📭 No development data found for {issue_key}")
+                return None
+
+        else:
+            logger.debug(
+                f"❌ Dev status API returned {response.status_code} for {issue_key}"
+            )
+            return None
+
+    except Exception as e:
+        logger.debug(f"💥 Error getting development status for {issue_key}: {str(e)}")
+        return None
+
+
+@jira_mcp.tool(tags={"jira", "read"})
+async def get_development_status(
+    ctx: Context,
+    issue_key: Annotated[str, Field(description="Jira issue key (e.g., 'SMS-8')")],
+    application_type: Annotated[
+        str,
+        Field(
+            description="Source control application type (e.g., 'GitHub', 'Bitbucket', 'GitLab')",
+            default="GitHub",
+        ),
+    ] = "GitHub",
+    data_type: Annotated[
+        str,
+        Field(
+            description="Type of development data to retrieve ('branch', 'pullrequest', 'commit')",
+            default="branch",
+        ),
+    ] = "branch",
+) -> str:
+    """Get development status information for a Jira issue including branch names, commits, and pull requests.
```

Comment:
The tool docstring should clearly warn about the experimental nature of this API. Add to docstring: "WARNING: This uses an experimental Jira API endpoint `/rest/dev-status/latest/issue/detail` that is not officially documented and may change without notice.